### PR TITLE
Documentation: Update Marker's main README

### PR DESCRIPTION
**Pull request by xFrednet**
What is Marker? Well, this! This repo! This project! This idea! Still not clear, well understandable. That's what README's are for. Allow me to introduce you to Marker's new business card. The first thing people will find when they stumble upon this project!

Okay, this makes the change sound more significant than it actually is. ^^ Anyways, I hope that this new README, gives a good first impression and defines what Marker actually is.

[:framed_picture: Rendered :framed_picture: ](https://github.com/xFrednet/marker/blob/151-this-is-readme/README.md)

---

A small but super important step for #151

Any feedback is very welcome!

### CODE REVIEW

**Comment by xFrednet** [comment]
Now I'm done with fixing all the nits I've noticed while reading it a second time. I like this new version :3

File: README.md [hunk 1]
```diff
@@ -1,53 +1,99 @@
...
+And more to come, see Marker's goals below.
+
+[Cargo]: https://github.com/rust-lang/cargo/
+<!-- FIXME(xFrednet): Update the link to link to `crates.io` once the first version was released-->
+[cargo_marker]: https://github.com/rust-marker/marker/tree/master/cargo-marker
+[rustc]: https://github.com/rust-lang/rustc/
+
+## Goals
+
+* **Stability**: Marker's API design heavily focuses on stability. Any lint, written after version 1.0.0, should compile and continue to work for years to come.
```
Comment (lucioleKi) [inline README.md:39]: Do you mean backward compatibility? The 'written after version 1.0.0' part can be made more clear.

File: README.md [hunk 2]
```diff
@@ -1,53 +1,99 @@
...
+4. Reach out to the wider community for more feedback.
+
+The used nightly version will be updated every 6 weeks, when a new version of Rust is released.
 
 ## Contributing
 
-Internal structure of marker and its components is documented in [`docs/internal/`](./docs/internal/).
+Contributions are highly appreciated! If you encounter any issues or have suggestions for improvements, please don't hesitate to [open an issue]. If you have an idea for a lint you want to implement with Marker, please share it by creating a [user story].
+
+Still reading? It would be wonderful if you're interested in helping out! Checkout Marker's [open issues] or [create a new one] to get suggestions for good first issues.
```
Comment (lucioleKi) [inline README.md:85]: How does [create a new one] provide suggestions for good first issues?

File: README.md [hunk 3]
```diff
@@ -1,53 +1,99 @@
...
+Marker is an experimental code analysis interface, with the goal to create a stable and user-friendly linting framework for the Rust programming language. Creating custom lints should be a straightforward process, and using them should be as simple as declaring a new dependency.
+
+Let's make custom lints and code analysis a reality!
 
 > **Note**
 >
-> The project is in the very early stages of development, some things are still missing and the API is not stable yet.
+> The project is in the early stages of development, some things are still missing and the API is not stable yet.
+>
+> A collection of limitations, goals and features is available below.
```
Comment (lucioleKi) [inline README.md:22]: Features, goals, and limitations (which is what you have later) is a better order.

File: README.md [hunk 4]
```diff
@@ -1,53 +1,99 @@
...
+
+## Features
+
+* **Custom Lints**: Marker provides a framework for everyone, to create and provide custom lints. Allowing you to automatically improve the code quality for your project and users of your crate.
```
Comment (lucioleKi) [inline README.md:26]: Marker offers a framework for everyone to create and provide custom lints, allowing you to automatically improve the code quality for your project.

File: README.md [hunk 5]
```diff
@@ -1,53 +1,99 @@
...
+* **User-Friendly Interface**: Marker provides a new subcommand for [Cargo] that does all the heavy lifting for you. [cargo_marker] can automatically set up a driver for linting, compile lint crate and run them on any project.
+* **Driver independent**: Every code analysis requires a driver that parses the code and provides further information. Marker's API is designed to be driver-independent. Allowing it to support future compilers and potentially IDEs, therefore binging custom lints everywhere. (Currently, only [rustc] is supported)
```
Comment (lucioleKi) [inline README.md:28]: Check grammar and logic for the last sentence.

**Review by lucioleKi — COMMENTED** [review]
Will make more comments tomorrow, mostly related to grammar and flow. Very good work already. I can see that you put in lots of effort here.

File: README.md [hunk 6]
```diff
@@ -1,53 +1,99 @@
...
-## Running
+## Limitations
 
-The project is currently only available in this GitHub repo.
-For a quick test, clone the repo and run `cargo dogfood`.
-This will start `cargo-marker`, load [`./marker_uitest`](./marker_uitest) as a *lint crate* and run it in this repository.
+Marker is still growing up, and that's a good thing. We can still shape the API and adapt it to what the user needs. However, this and the fact that Marker is not an official Rust project comes with some limitations:
+
+* **Nightly**: Internally, Marker has to rely on nightly versions of rustc. However, you, as an end-user, should not notice this dependency.
+* **Lint Crates Registry**: Lint creates can currently only be compiled from paths and git repos. Fetching lint crates from [crates.io] is planned, but not yet supported.
```
Comment (lucioleKi) [inline README.md:60]: creates -> crates typo here

File: README.md [hunk 7]
```diff
@@ -1,53 +1,99 @@
...
+* **AST Limitations**: Marker's API is still missing a few elements to represent everything required for linting. The API is still incomplete when it comes to:
+    * `async` expressions
+    * Higher order types
+    * Array types
+    * Attributes
+    * Macros
+* **Utility**: The API mostly focussed on representing everything in Rust. Next, the API needs to be extended with more utility functions, to make life easier for everyone.
```
Comment (lucioleKi) [inline README.md:67]: Could be more specific

File: README.md (hunk 4 above)
Comment (xFrednet) [reply README.md:26]: I want to keep the point, that lints can be provided for other users, highlighted. I like the rewrite of the first part :+1: Now I've written:

* **Custom Lints**: Marker offers a framework for everyone to create and provide custom lints, allowing you to automatically improve the code quality for you and users of your crate.

File: README.md (hunk 1 above)
Comment (xFrednet) [reply README.md]: Do you think the updated version is better? :)

**Comment by xFrednet** [comment]
I'm now happy with this version. A big thank you to my friends that I bothered with proofreading :heart: 

bors r=lucioleKi

**Pull request merged**